### PR TITLE
Change FeatureFlipper to use `hostname` from settings.local.yml to de…

### PR DESCRIPTION
**Pull request by jholton**
## Description of change
Change FeatureFlipper to use `hostname` from settings.local.yml to determine if we are in a non-production environment, instead of the GovDelivery URL.  See [issue](https://github.com/department-of-veterans-affairs/vets.gov-team/issues/18010) for motivation.

The `hostname` values for the various environments are:
`api.vets.gov`
`staging-api.vets.gov`
`dev-api.vets.gov`

## Testing done
specs, manual testing in `dev`

## Testing planned
manual testing in other environments

#### Unique to this PR

- [x] Change `FeatureFlipper.staging_email?` to use `hostname` from settings.local.yml

#### Applies to all PRs

- ~~[ ] Appropriate logging~~
- ~~[ ] Swagger docs have been updated, if applicable~~
- [x] Provide link to originating GitHub issue, or connected to it via ZenHub
- [x] Does not contain any sensitive information (i.e. PII/credentials/internal URLs/etc., in logging, hardcoded, or in specs)
- ~~[ ] Provide which alerts would indicate a problem with this functionality (if applicable)~~

### CODE REVIEW

File: lib/feature_flipper.rb [hunk 1]
```diff
@@ -6,7 +6,7 @@ def self.send_email?
   end
 
   def self.staging_email?
-    Settings.reports.server.include?('stage')
+    (Settings.hostname =~ /-api/).present?
```
Comment (wyattwalter) [inline lib/feature_flipper.rb]: Can we just set this to the value that it should be in settings.local.yml instead of checking hostnames?

Comment (jholton) [reply lib/feature_flipper.rb]: Sure.  What I'm really looking for here is a flag to tell if we're in production or not. That'll probably be useful in more than one situation.
I thought using the hostname would be a good way to do that, but this is susceptible to changes to the host subdomain.
To be even more future-proof, what if we add an entry to settings.local.yml that just lists the name of the deployed environment?

Comment (kfrz) [reply lib/feature_flipper.rb]: FWIW this has been brought up before - https://github.com/department-of-veterans-affairs/vets-api/pull/2664

Comment (hpjaj) [reply lib/feature_flipper.rb]: +1 to implement #2664

Comment (jholton) [reply lib/feature_flipper.rb]: Since that conversation has already been had, I'm going to:
1. Close this
2. Open a devops PR
3. Open a new API PR.

**Pull request closed without merging**